src/main.c

```diff
@@ -13,6 +15,23 @@
 
 int main(int argc, char **argv)
 {
+    char input[50];
+    char loopEndstr[] = "quit";
+    int  endLoop      = false;
+
+    while (!endLoop) {
+        fgets(input, sizeof(input), stdin);
+        input[strlen(input) - 1] = '\0';
+        endLoop = strcmp(input, loopEndstr) == 0 ? true : false;
+        // printf("in side the loop endloop : %d \n", endLoop);
+
+        if (endLoop) {
+            printf("program is gone quit");
+        } else {
+            printf(" %s,\ninputlen : %lu \n", input, strlen(input));
+        }
```

Comment:
```c
if (strncmp(input, quit, 5) == 0)
    break;
else
    puts(input);
```
Always use the safe size comparators and other functions where you check the size we use 5 not 4 (the size of the word quit) because we check for the null terminator or else this would match anything that starts with quit